perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh

```diff
@@ -1,69 +1,103 @@
-BUCKET_NAME=presubmit-perf-tests
-MOUNT_POINT=gcs
-# The VM will itself exit if the gcsfuse mount fails.
-CGO_ENABLED=0 go run . $GCSFUSE_FLAGS $BUCKET_NAME $MOUNT_POINT
-touch result.txt
-# Running FIO test
-chmod +x perfmetrics/scripts/presubmit/run_load_test_on_presubmit.sh
-./perfmetrics/scripts/presubmit/run_load_test_on_presubmit.sh
-sudo umount gcs
-
 # Fetch PR branch
 echo '[remote "origin"]
          fetch = +refs/pull/*/head:refs/remotes/origin/pr/*' >> .git/config
-git fetch origin
-echo checkout PR branch
-git checkout pr/$KOKORO_GITHUB_PULL_REQUEST_NUMBER
+git fetch origin -q
+
+function execute_perf_test() {
+  mkdir -p gcs
+  GCSFUSE_FLAGS="--implicit-dirs --max-conns-per-host 100"
+  BUCKET_NAME=presubmit-perf-tests
+  MOUNT_POINT=gcs
+  # The VM will itself exit if the gcsfuse mount fails.
+  go run . $GCSFUSE_FLAGS $BUCKET_NAME $MOUNT_POINT
+  # Running FIO test
+  chmod +x perfmetrics/scripts/presubmit/run_load_test_on_presubmit.sh
+  ./perfmetrics/scripts/presubmit/run_load_test_on_presubmit.sh
+  sudo umount gcs
+}
+
+# execute perf tests.
+if [[ "$perfTestStr" == *"$EXECUTE_PERF_TEST_LABEL"* ]];
+then
+ # Installing requirements
+ echo Installing python3-pip
+ sudo apt-get -y install python3-pip
+ echo Installing libraries to run python script
+ pip install google-cloud
+ pip install google-cloud-vision
+ pip install google-api-python-client
+ pip install prettytable
+ echo Installing fio
+ sudo apt-get install fio -y
 
-# Executing integration tests
-GODEBUG=asyncpreemptoff=1 CGO_ENABLED=0 go test ./tools/integration_tests/... -p 1 --integrationTest -v --testbucket=gcsfuse-integration-test -timeout 24m
+ # Executing perf tests for master branch
+ git checkout master
+ # Store results
+ touch result.txt
+ echo Mounting gcs bucket for master branch and execute tests
+ execute_perf_test
 
-# Executing perf tests
-echo Mounting gcs bucket from pr branch
-mkdir -p gcs
-# The VM will itself exit if the gcsfuse mount fails.
-CGO_ENABLED=0 go run . $GCSFUSE_FLAGS $BUCKET_NAME $MOUNT_POINT
-# Running FIO test
-chmod +x perfmetrics/scripts/presubmit/run_load_test_on_presubmit.sh
-./perfmetrics/scripts/presubmit/run_load_test_on_presubmit.sh
-sudo umount gcs
 
-echo showing results...
-python3 ./perfmetrics/scripts/presubmit/print_results.py
+ # Executing perf tests for PR branch
+ echo checkout PR branch
+ git checkout pr/$KOKORO_GITHUB_PULL_REQUEST_NUMBER
+ echo Mounting gcs bucket from pr branch and execute tests
+ execute_perf_test
+
+ # Show results
+ echo showing results...
+ python3 ./perfmetrics/scripts/presubmit/print_results.py
+fi
+
+# Execute integration tests.
+if [[ "$integrationTestsStr" == *"$EXECUTE_INTEGRATION_TEST_LABEL"* ]];
+then
+  echo checkout PR branch
+  git checkout pr/$KOKORO_GITHUB_PULL_REQUEST_NUMBER
+
+  # Create bucket for integration tests.
+  # The prefix for the random string
+  bucketPrefix="gcsfuse-integration-test-"
+  # The length of the random string
+  length=5
```

Comment:
Do we have any issue in using larger random string? If not we can increase the length upto 10 or 20. Just to reduce the probability further to avoid collision.

Reply:
Current # of combinations is 36^5 ~= 60M . It's a high number technically, and has little chance of repeating. Making it length 8 or more would make the chances of collision almost impossible (2.8 trillion combinations).  

There are restrictions with GCS bucket names given at https://cloud.google.com/storage/docs/buckets#naming which could be taken care of to avoid failures in bucket creation, but I'd suggest to take that up later.

Reply:
Also at a time only two kokoro can build and, I guess this script is getting used by kokoro pr perf tests. so thare are very very rare chances of collision probability. So I think 5 is good number.